### PR TITLE
Update Unit Locate instruction to access located building

### PR DESCRIPTION
**Pull request by z0mbiesrock**
Update the instruction so that when a "building" is specified for locating, it also returns the building itself to allow access to its variables and as a variable. This would allow a controlled unit to directly interact with the block

An example of a logic process that would work from this change is:
```
 ubind @nova
 sensor cnt @unit @totalItems
 sensor icp @unit @itemCapacity
 ucontrol boost 1 0 0 0
 ulocate building core false @copper creX creY found cre
 ucontrol approach creX creY 10 0
 ucontrol itemTake cre @silicon icp 0
```

The current unitLocate command doesn't return the block; currently the nova would just hover endlessly over the core through this process and take no items whatsoever.

image
I'm certain this would probably get rejected, because I am uncertain as to why

### CODE REVIEW

**Comment by z0mbiesrock** [comment]
RIP. A syntax error has spiked the chances of rejection to 100%

File: core/src/mindustry/logic/LStatements.java [hunk 1]
```diff
@@ -905,6 +906,12 @@ void rebuild(Table table){
             table.add(" found ").left();
             fields(table, outFound, str -> outFound = str);
 
+            if(locate == building) {
```
Comment (summetdev) [inline core/src/mindustry/logic/LStatements.java:909]:
```suggestion
            if(locate == building){
```

**Comment by summetdev** [comment]
> RIP. A syntax error has spiked the chances of rejection to 100%

Use an IDE like IntelliJ Idea to avoid syntax errors

**Comment by Anuken** [comment]
> because I am uncertain as to why

There are many issues that could occur:
- controlling/reading data from blocks not on your team
- controlling blocks far away from the processor with no visible connection
- saving a reference to a building and controlling it even when the unit near it is gone
- saving and controlling a building which no longer exists/has been destroyed

I will implement this myself when I need to.

**Pull request closed without merging**